src/schemas/__tests__/licenses.schema.unit.spec.js

```diff
@@ -47,4 +48,63 @@ describe('licenses.schema.unit', () => {
       expect(error.details[0].message).toContain('"licence" must be a string')
     })
   })
+
+  describe('fullLicenceLoginRequestParamSchema', () => {
+    it('should validate successfully when the "licence" field is provided and matches the pattern', () => {
+      const params = { licence: 'ABC123-XYZ' }
+      const { error } = fullLicenceLoginRequestParamSchema.validate(params)
+
+      expect(error).toBeUndefined()
+    })
+
+    it('should return an error if the "licence" field is missing', () => {
+      const params = {}
+      const { error } = fullLicenceLoginRequestParamSchema.validate(params)
+
+      expect(error).toBeDefined()
+      expect(error.details[0].message).toContain('"licence" is required')
+    })
+
+    it('should return an error if the "licence" field is not a string', () => {
+      const params = { licence: 123456 }
+      const { error } = fullLicenceLoginRequestParamSchema.validate(params)
+
+      expect(error).toBeDefined()
+      expect(error.details[0].message).toContain('"licence" must be a string')
+    })
+
+    it('should return an error if the "licence" field contains an underscore', () => {
+      const params = { licence: 'ABC_123' }
+      const { error } = fullLicenceLoginRequestParamSchema.validate(params)
+
+      expect(error).toBeDefined()
+      expect(error.details[0].message).toContain(
+        '"licence" with value "ABC_123" fails to match the required pattern'
+      )
+    })
+
+    it('should return an error if the "licence" field contains special characters other than hyphen', () => {
+      const params = { licence: 'ABC@123' }
+      const { error } = fullLicenceLoginRequestParamSchema.validate(params)
+
+      expect(error).toBeDefined()
+      expect(error.details[0].message).toContain(
+        '"licence" with value "ABC@123" fails to match the required pattern'
+      )
+    })
+
+    it('should validate successfully when the "licence" field contains only letters and numbers', () => {
+      const params = { licence: 'ABC123' }
+      const { error } = fullLicenceLoginRequestParamSchema.validate(params)
+
+      expect(error).toBeUndefined()
+    })
+
+    it('should validate successfully when the "licence" field contains letters, numbers, and hyphens', () => {
+      const params = { licence: '23210126-2WC3FBP-ABNFA7' }
+      const { error } = fullLicenceLoginRequestParamSchema.validate(params)
+
+      expect(error).toBeUndefined()
+    })
```

Comment:
This is another identical test - so combine it with the previous test and the first test in an `each`